hypothesis-python/tests/typing_extensions/test_backported_types.py

```diff
@@ -202,3 +204,31 @@ def test_callable_return_typegard_type():
 
     with pytest.raises(InvalidArgument, match="Cannot register generic type"):
         st.register_type_strategy(Callable[[], TypeGuard[int]], st.none())
+
+
+class Movie(TypedDict):  # implicitly total=True
+    title: str
+    year: NotRequired[int]
+
+
+@given(from_type(Movie))
+def test_typeddict_not_required(value):
+    assert type(value) == dict
+    assert set(value).issubset({"title", "year"})
+    assert isinstance(value["title"], str)
+    if "year" in value:
+        assert isinstance(value["year"], int)
+
+
+class OtherMovie(TypedDict, total=False):
+    title: Required[str]
+    year: int
+
+
+@given(from_type(OtherMovie))
+def test_typeddict_required(value):
+    assert type(value) == dict
+    assert set(value).issubset({"title", "year"})
+    assert isinstance(value["title"], str)
+    if "year" in value:
+        assert isinstance(value["year"], int)
```

Comment:
These are good tests, but I'd like to add some more:

- Check that we can generate an example that does have a year, and an example that does *not* have a year
- Check behaviour when you mix inheritance-based and pep655-based specs: we should be able to use `total=...` for some keys and have that respected, and also overwrite that using pep655.

```python
class Story(TypedDict, total=True):
    author: str

class Book(Story, total=False):
    pages: int

class Novel(Book):
    genre: Required[str]
    rating: NotRequired[str]

# Check that we can generate a novel with author & genre only;
# author & genre & pages, author & genre & rating, and all four keys
```

(also this is a terrible terrible data model, please feel free to replace it)

Reply:
it works, tests added